src/main/java/com/midaswebserver/midasweb/services/UserServiceImp.java

```diff
@@ -0,0 +1,264 @@
+     */
+    @Override
+    public boolean add(User user) {
+        if (user == null) {
+            log.error("add: User is null");
+            return false;
+        }
+        if (user.getUsername()==null ||  user.getEmail()==null || user.getPhoneNumber()==null || user.getHashedPassword()==null) {
+            log.error("add: One of fields in user found to be null");
+            return false;
+        }
+        if (user.getUsername().isBlank() ||  user.getEmail().isBlank() || user.getPhoneNumber().isBlank() || user.getHashedPassword().isBlank()) {
+            log.error("add: One of fields in user found to be blank");
+            return false;
+        }
+        if (!validateUniqueUsername(user.getUsername())) {
+            log.error("add: '{}' has the same name in database", user.getUsername());
+            return false;
+        }
+        userRepo.save(user);
+        log.debug("add: '{}' has been added to the database", user.getUsername());
+        return true;
+    }
+
+    /**
+     * ValidateUniqueUsername checks to see if there is another person in the database with the same username
+     *
+     * @param username
+     * @return boolean
+     */
+    @Override
+    public boolean validateUniqueUsername(String username) {
+        log.debug("validateUser: '{}' attempted login", getUserByUsername(username));
+        // Always do the lookup in a case-insensitive manner (lower-casing the data).
+        List<User> users = userRepo.findByUsernameIgnoreCase(username);
+
+        // We expect 0 or 1, so if we get more than 1, bail out as this is an error we don't deal with properly.
+        if (users.size() != 0) {
+            log.debug("validateUniqueUsername: found {} users, invalid quantity. Valid = 1", users.size());
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * delete deletes a user from the database
+     *
+     * @param user
+     * @return boolean
+     */
+    @Override
+    public boolean delete(User user) {
+        if (user == null) {
+            log.error("Delete: null was passed to method");
+            return false;
+        }
+        if (this.getUserByUsername(user.getUsername()) == null) {
+            log.error("Delete: User '{}' was not found in database to delete", user.getUsername());
+            return false;
+        }
+        userRepo.delete(user);
+        log.debug("Delete: User '{}' was deleted from the database", user.getUsername());
+        return true;
+    }
+
+
+    /**
+     * Gets User by Id
+     * TODO have validation and safe and expected returns;
+     *
+     * @param ID
+     * @return User if user excists, returns null otherwise and if user passes null ID
+     */
+    @Override
+    public User getUserById(Long ID) {
+        if (ID == null) {
+            return null;
+        }
+        try {
+            return userRepo.findById(ID).get();
+        } catch (Exception e) {
```

Comment:
Don't catch 'Exception', determine which error will be thrown and catch the specific exception you expect to be handle.